lib/esbonio/esbonio/sphinx_agent/handlers/directives.py

```diff
@@ -0,0 +1,104 @@
+import importlib
+import inspect
+from typing import List
+from typing import Optional
+from typing import Type
+
+from docutils.parsers.rst import Directive
+from docutils.parsers.rst import directives as docutils_directives
+
+from .. import types
+from ..app import Database
+from ..app import Sphinx
+from ..util import as_json
+
+DIRECTIVES_TABLE = Database.Table(
+    "directives",
+    [
+        Database.Column(name="name", dtype="TEXT"),
+        Database.Column(name="implementation", dtype="TEXT"),
+        Database.Column(name="location", dtype="JSON"),
+    ],
+)
+
+
+def get_impl_name(directive: Type[Directive]) -> str:
+    try:
+        return f"{directive.__module__}.{directive.__name__}"
+    except AttributeError:
+        return f"{directive.__module__}.{directive.__class__.__name__}"
+
+
+def get_impl_location(impl: Type[Directive]) -> Optional[str]:
```

Comment:
This is too slow to compute eagerly.... :thinking:

Reply:
Disabled for now, it will be interested to see if we can compute this on demand as the agent discovers directive usage via symbols....